### ISSUE TITLE
Issue in training the code2vec

### ISSUE THREAD

**Issue by faysalhossain2007** [tech-srl/code2vec#101]
Thanks for the interesting project!

I am trying to run the code2vec train file. But I am getting error everytime and I am not able to see the console output because as soon as the error occurs the console automatically shuts off.

I am running on Ubuntu. I am using conda for managing the dependencies.

`Python - Python 3.6.12 :: Anaconda, Inc.`

`Tensorflow 2.0.0`

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Thu_Jun_11_22:26:38_PDT_2020
Cuda compilation tools, release 11.0, V11.0.194
Build cuda_11.0_bu.TC445_37.28540450_0
```

I am using the preprocessed data downloaded using the given instructions. 

```
wget https://s3.amazonaws.com/code2vec/data/java14m_data.tar.gz
tar -xvzf java14m_data.tar.gz
```

Btw, I am able to run the evaluation code: 

`python3 code2vec.py --load models/java14_model/saved_model_iter8.release --test data/java14m/java14m.test.c2v
`
which shows me the following performance 
 
```
top10_acc: [0.50086444 0.54063157 0.5556596  0.56424161 0.56990053 0.57424853
 0.57761945 0.58042313 0.58295811 0.58502626], precision: 0.6582174600509684, recall: 0.5314902438062055, F1: 0.5881043842714981
```

**Comment by urialon** [comment]
Hi @faysalhossain2007 ,
Maybe you can try to run it and redirect the output to a text file, so even if the terminal closes, you can recover the error?

something like:
`python3 code2vec.py ... > mylog.txt`

**Comment by faysalhossain2007** [comment]
My training script file gives me the following output and closes the terminal: 

```
2020-12-08 13:36:32,177 INFO     
2020-12-08 13:36:32,177 INFO     
2020-12-08 13:36:32,177 INFO     ---------------------------------------------------------------------
2020-12-08 13:36:32,177 INFO     ---------------------------------------------------------------------
2020-12-08 13:36:32,177 INFO     ---------------------- Creating code2vec model ----------------------
2020-12-08 13:36:32,177 INFO     ---------------------------------------------------------------------
2020-12-08 13:36:32,177 INFO     ---------------------------------------------------------------------
2020-12-08 13:36:32,177 INFO     Checking number of examples ...
```

**Comment by urialon** [comment]
Hi ,
What is the command line that you ran?

**Comment by faysalhossain2007** [comment]
`source train.sh`

And I have just modified the following code: 
`python3 -u code2vec.py --data ${data} --test ${test_data} --save ${model_dir}/saved_model > log_error.txt`

**Comment by faysalhossain2007** [comment]
Btw I have also changed the config.py. Do you think that is causing the problem? I have added the configuration here:


    ```
def set_defaults(self):
        self.NUM_TRAIN_EPOCHS = 5 # 20
        self.SAVE_EVERY_EPOCHS = 1
        self.TRAIN_BATCH_SIZE = 512 #1024
        self.TEST_BATCH_SIZE = self.TRAIN_BATCH_SIZE
        self.TOP_K_WORDS_CONSIDERED_DURING_PREDICTION = 10
        self.NUM_BATCHES_TO_LOG_PROGRESS = 100
        self.NUM_TRAIN_BATCHES_TO_EVALUATE = 1800
        self.READER_NUM_PARALLEL_BATCHES = 6  # cpu cores [for tf.contrib.data.map_and_batch() in the reader]
        self.SHUFFLE_BUFFER_SIZE = 10000
        self.CSV_BUFFER_SIZE = 100 * 1024 * 1024  # 100 MB
        self.MAX_TO_KEEP = 10

        # model hyper-params
        self.MAX_CONTEXTS = 200
        self.MAX_TOKEN_VOCAB_SIZE = 10000 #1301136
        self.MAX_TARGET_VOCAB_SIZE = 5000 #261245
        self.MAX_PATH_VOCAB_SIZE = 5000 #911417
        self.DEFAULT_EMBEDDINGS_SIZE = 128
        self.TOKEN_EMBEDDINGS_SIZE = self.DEFAULT_EMBEDDINGS_SIZE
        self.PATH_EMBEDDINGS_SIZE = self.DEFAULT_EMBEDDINGS_SIZE
        self.CODE_VECTOR_SIZE = self.context_vector_size
        self.TARGET_EMBEDDINGS_SIZE = self.CODE_VECTOR_SIZE
        self.DROPOUT_KEEP_RATE = 0.75
        self.SEPARATE_OOV_AND_PAD = False
```

**Comment by faysalhossain2007** [comment]
This is the output of the preprocess.sh:

```
Extracting paths from validation set...
Finished extracting paths from validation set
Extracting paths from test set...
Finished extracting paths from test set
Extracting paths from training set...
Finished extracting paths from training set
Creating histograms from the training data
File: java-small-preprocessed.test.raw.txt
Average total contexts: 224.2141725273747
Average final (after sampling) contexts: 90.19777441467106
Total examples: 56165
Empty examples: 0
Max number of contexts per word: 13272
File: java-small-preprocessed.val.raw.txt
Average total contexts: 166.43046160391407
Average final (after sampling) contexts: 70.6427568602425
Total examples: 23505
Empty examples: 0
Max number of contexts per word: 47830
File: java-small-preprocessed.train.raw.txt
Average total contexts: 164.10066078798403
Average final (after sampling) contexts: 77.02187892319374
Total examples: 665115
Empty examples: 0
Max number of contexts per word: 59724
Dictionaries saved to: data/java-small-preprocessed/java-small-preprocessed.dict.c2v

```

**Comment by faysalhossain2007** [comment]
this problem is solved! Thanks for sharing the tool with all! I appreciate it.

**Comment by urialon** [comment]
Great! I'm glad to hear.
For future reference, can you share what was the problem?

**Comment by faysalhossain2007** [comment]
Pointing wrong dataset path and also an extra space in the script file.

**Comment by Avra2** [comment]
> Btw I have also changed the config.py. Do you think that is causing the problem? I have added the configuration here:
> 
> > def set_defaults(self):
> > self.NUM_TRAIN_EPOCHS = 5 # 20
> > self.SAVE_EVERY_EPOCHS = 1
> > self.TRAIN_BATCH_SIZE = 512 #1024
> > self.TEST_BATCH_SIZE = self.TRAIN_BATCH_SIZE
> > self.TOP_K_WORDS_CONSIDERED_DURING_PREDICTION = 10
> > self.NUM_BATCHES_TO_LOG_PROGRESS = 100
> > self.NUM_TRAIN_BATCHES_TO_EVALUATE = 1800
> > self.READER_NUM_PARALLEL_BATCHES = 6  # cpu cores [for tf.contrib.data.map_and_batch() in the reader]
> > self.SHUFFLE_BUFFER_SIZE = 10000
> > self.CSV_BUFFER_SIZE = 100 * 1024 * 1024  # 100 MB
> > self.MAX_TO_KEEP = 10
> > ```
> >     # model hyper-params
> >     self.MAX_CONTEXTS = 200
> >     self.MAX_TOKEN_VOCAB_SIZE = 10000 #1301136
> >     self.MAX_TARGET_VOCAB_SIZE = 5000 #261245
> >     self.MAX_PATH_VOCAB_SIZE = 5000 #911417
> >     self.DEFAULT_EMBEDDINGS_SIZE = 128
> >     self.TOKEN_EMBEDDINGS_SIZE = self.DEFAULT_EMBEDDINGS_SIZE
> >     self.PATH_EMBEDDINGS_SIZE = self.DEFAULT_EMBEDDINGS_SIZE
> >     self.CODE_VECTOR_SIZE = self.context_vector_size
> >     self.TARGET_EMBEDDINGS_SIZE = self.CODE_VECTOR_SIZE
> >     self.DROPOUT_KEEP_RATE = 0.75
> >     self.SEPARATE_OOV_AND_PAD = False
> > ```
> 
> I have preprocessed java small dataset by myself. In my preprocessed folder I have following four files:
> 
> `java-small-preprocessed.dict.c2v, java-small-preprocessed.test.c2v, java-small-preprocessed.train.c2v, java-small-preprocessed.val.c2v `

Can you please post the output of `dict.c2v` here?
